specification/keyvault/data-plane/TypeSpec/routes.tsp

```diff
@@ -0,0 +1,278 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/rest";
+import "./models.tsp";
+
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+alias KeyVaultOperations = StandardResourceOperations;
+
+alias KeyVaultListParameters = ListQueryParametersTrait<{
+  /**
+   * Maximum number of results to return in a page. If not specified the service
+   * will return up to 25 results.
+   */
+  @maxValue(25)
+  @minValue(1)
+  @query("maxresults")
+  maxresults?: int32;
+}>;
+
+/**
+ *  The SET operation adds a secret to the Azure Key Vault. If the named secret
+ * already exists, Azure Key Vault creates a new version of that secret. This
+ * operation requires the secrets/set permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Sets a secret in a specified key vault.")
+@route("/secrets/{secretName}")
+@put
+op setSecret is Azure.Core.Foundations.Operation<
```

Comment:
Why aren't these models as resource collections? My understanding is that we shouldn't use Foundations unless necessary. "secrets" are a collection, and "version" is a sub-resource of a "secrets/{name}" collection.

Reply:
The reason that this operation -- and many others that are resource operations conceptually -- is using `Foundations.Operation` is because KV models tend to have an `id` property in place of a resource name. This ID is a full URL and can't be used as the key for resource operations that follow the standard template, which is why the only standard resource operations that I was able to use were list operations, since they're just to a route like `/secrets` and don't run into this mismatching key issue.

Reply:
If there's a way to create a `name` property that somehow parses out the value from `id`, we could make many more of these operations use standard templates. But my understanding was that there's an incompatibility because of how the models are specified by the service.